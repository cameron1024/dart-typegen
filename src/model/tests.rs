use super::*;

macro_rules! parse_snapshot {
    ($name:ident) => {
        #[test]
        fn $name() {
            let text = include_str!(crate::test_file!($name));
            let config = Library::parse_impl(None, text).unwrap();

            insta::assert_debug_snapshot!(config);
        }
    };
}

crate::all_test_files!(parse_snapshot);
