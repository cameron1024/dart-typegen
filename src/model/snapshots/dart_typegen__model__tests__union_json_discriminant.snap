---
source: src/model/tests.rs
expression: config
---
Library {
    preamble: None,
    postamble: None,
    defaults: None,
    enums: [],
    classes: [],
    unions: [
        Union {
            span: Span(
                0,
                0,
            ),
            name: SpannedScalar {
                value: "Animal",
                span: SourceSpan {
                    offset: SourceOffset(
                        53,
                    ),
                    length: 8,
                },
            },
            sealed: None,
            json_discriminant: Some(
                SpannedScalar {
                    value: "__type",
                    span: SourceSpan {
                        offset: SourceOffset(
                            84,
                        ),
                        length: 8,
                    },
                },
            ),
            docs: None,
            classes: [
                Class {
                    span: Span(
                        0,
                        0,
                    ),
                    name: SpannedScalar {
                        value: "Dog",
                        span: SourceSpan {
                            offset: SourceOffset(
                                102,
                            ),
                            length: 5,
                        },
                    },
                    fields: [],
                    docs: None,
                    extra_dart: [],
                },
                Class {
                    span: Span(
                        0,
                        0,
                    ),
                    name: SpannedScalar {
                        value: "Cat",
                        span: SourceSpan {
                            offset: SourceOffset(
                                116,
                            ),
                            length: 5,
                        },
                    },
                    fields: [],
                    docs: None,
                    extra_dart: [],
                },
            ],
        },
    ],
}
